pkg/operator/sync.go

```diff
@@ -206,15 +227,23 @@ func (optr *Operator) syncRenderConfig(_ *renderConfig) error {
 	}
 	spec.AdditionalTrustBundle = trustBundle
 
-	// if the cloudConfig is set in infra read the cloud config reference
-	if infra.Spec.CloudConfig.Name != "" {
-		cc, err := optr.getCloudConfigFromConfigMap("openshift-config", infra.Spec.CloudConfig.Name, infra.Spec.CloudConfig.Key)
+	// check if openshift-config-managed/kube-cloud-config ConfigMap is available and cloud config is required for given platform
```

Comment:
personally, instead of a shallow function like processCloudConfigMapError, i think a function that handled cloud.conf would be better because that keeps the syncRenderConfig simple and also improves the readability.

Reply:
agree. lots of `if` and error returns made me play extra safe :)
moved everything now in syncCloudConfig() function.